go/stats/counters.go

```diff
@@ -143,14 +110,15 @@ type CountersWithSingleLabel struct {
 func NewCountersWithSingleLabel(name, help, label string, tags ...string) *CountersWithSingleLabel {
 	c := &CountersWithSingleLabel{
 		counters: counters{
-			counts: make(map[string]*int64),
+			counts: make(map[string]int64),
 			help:   help,
 		},
-		label: label,
+		label:        label,
+		labelDropped: IsDimensionDropped(label),
 	}
 
 	for _, tag := range tags {
-		c.counts[tag] = new(int64)
+		c.counts[tag] = 0
```

Comment:
We shouldn't preinitialize the zero values in the case where that dimension is dropped.